### ISSUE TITLE
False positive asserting after continue

### ISSUE THREAD

**Issue by muglug** [vimeo/psalm#1000]
```php
$i = 0;

foreach ([rand(0, 1)] as $n) {
    $i = $n;
  
    if ($i !== 0) {
        continue;
    }
}

if ($i !== 0) {}
```
Expected: No issue
Actual: https://getpsalm.org/r/9ae460759f

cc @wirone
